jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/execution/PathMapperTest.kt

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.nio.file.Files
+
+class PathMapperTest {
+    @Rule
+    @JvmField
+    val tempFolder = TemporaryFolder()
+
+    private lateinit var mapper: PathMapper
+
+    @Test
+    fun firstMatchWinsLocally() {
+        mapper {
+            addMapping("local", "remote")
+            addMapping("local2", "remote2")
+            addMapping("local2", "remote2/sub/")
+        }
+
+        createLocalFile("local2/foo")
+
+        assertThat(convertToLocal("remote2/foo")).isEqualTo("local2/foo")
+    }
+
+    @Test
+    fun firstMatchWinsRemotely() {
+        mapper {
+            addMapping("local", "remote")
+            addMapping("local2", "remote2")
+            addMapping("local2", "remote2/sub/")
+        }
+
+        assertThat(mapper.convertToRemote(createLocalFile("local2/foo"))).isEqualTo("remote2/foo")
+    }
+
+    @Test
+    fun onlyPrefixIsReplaced() {
+        mapper {
+            addMapping("local/sub", "remote/sub/sub")
+        }
+
+        assertThat(mapper.convertToRemote(createLocalFile("local/sub/foo"))).isEqualTo("remote/sub/sub/foo")
+    }
+
+    @Test
+    fun matchesAtFolderBoundary() {
+        mapper {
+            addMapping("local", "remote")
+            addMapping("localFolder", "remote2")
+        }
+
+        assertThat(mapper.convertToRemote(createLocalFile("localFolder/foo"))).isEqualTo("remote2/foo")
+    }
+
+    @Test
+    fun fileMustExistLocally() {
+        mapper {
+            addMapping("local1", "remote")
+            addMapping("local2", "remote")
+        }
+
+        createLocalFile("local2/foo")
+
+        assertThat(convertToLocal("remote/foo")).isEqualTo("local2/foo")
+    }
+
+    @Test
+    fun trailingSlashIsIgnored() {
+        mapper {
+            addMapping("local/", "remote/")
+            addMapping("local2", "remote2")
+        }
+
+        assertThat(mapper.convertToRemote(createLocalFile("local/foo"))).isEqualTo("remote/foo")
+        assertThat(convertToLocal("remote/foo")).isEqualTo("local/foo")
+
+        assertThat(mapper.convertToRemote(createLocalFile("local2/foo"))).isEqualTo("remote2/foo")
+        assertThat(convertToLocal("remote2/foo")).isEqualTo("local2/foo")
+    }
+
+    private fun convertToLocal(remote: String) = mapper.convertToLocal(remote)
+        ?.removePrefix(tempFolder.root.absolutePath)
+        ?.removePrefix("/")
+
+    private fun createLocalFile(path: String): String {
+        val file = tempFolder.root.toPath().resolve(path)
+        Files.createDirectories(file.parent)
+        if (!Files.exists(file)) {
+            Files.createFile(file)
+        }
+
+        return file.toString()
+    }
+
+    private fun mapper(init: MutableList<PathMapping>.() -> Unit) {
```

Comment:
nit: Can you rename this method to something like `initMapper`, it is a bit confusing to have the same name to the member variable.

Reply:
Yep